Implemented slice-based writing of binary data

rreilink opened the pull request:
Implemented slice-based writing of binary data using variable[start:end]=b'data'

Implemented support for variable types with dummy space at the end. This removes several 'Ctype size of datatype %s does not match, replaced with dummy ctype of %d bytes' warnings

rreilink committed:
Implemented slice-based writing of binary data
Implemented support for variable types with dummy space at the end

T-BA commented:
Hi Rob, I will take a look at your changes. 

Can it only be used to write data? Would it be possible / useful to add reading as well? Mainly to keep the API consistent, otherwise sometimes slicing does work and sometimes it does not, right?

T-BA commented:
Some intital findings, please let me know if I try to use it wrongly.

As a test project I added a struct to a plc project:
TYPE myStruct :
STRUCT
	iOne: SINT;
	iTwo: INT;
	iFour: DINT;
	iEight: LINT;
END_STRUCT
END_TYPE

Then from python:
from ads import adssymbols
v = adssymbols.getVariables()

1) default values for start and end are not recognised:
v.MAIN.myStruct.iFour[:] = b'datatata'
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\GIT\python-ads\ads\adssymbols.py", line 502, in __setitem__
    assert start >= 0
TypeError: '>=' not supported between instances of 'NoneType' and 'int'

2) It seems that writing does not work for me? What's going wrong here?
v.MAIN.myStruct.iFour[1:3] = b'datatata'
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\GIT\python-ads\ads\adssymbols.py", line 505, in __setitem__
    cpyads.adsSyncWriteReq(self.__vardef.amsAddress, self.__symbol.iGroup, self.__symbol.iOffs + self.__offset + start, (cpyads.c_ubyte * (stop-start)).from_buffer(data))
TypeError: underlying buffer is not writable

rreilink committed:
- Implemented default indexing ([:], [:n], [n:]) for reading and writing raw data

- Fixed the issue that writing raw binary data did not work if the data argument was a read-only buffer

rreilink commented:
Hi Teun,

Thanks for your review comments. I have addressed them in the latest commit

Note that it should be v.MAIN.myStruct.iFour[:] = b'data' since iFour[:] is four bytes in size. A descriptive ValueError is raised in this case now.

T-BA reviewed (approved):
@rreilink : Thank you for addressing my comments and making the ValueError clearer.

@timonmentink , can you merge this branch?
